modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Traverse.kt

```diff
@@ -5,11 +5,519 @@ import arrow.core.Id
+ *
+ * fun parseIntEither(s: String): Either<NumberFormatException, Int> =
+ *   if (s.matches(Regex("-?[0-9]+"))) Either.Right(s.toInt())
+ *   else Either.Left(NumberFormatException("$s is not a valid integer."))
+ *
+ * fun parseIntValidated(s: String): ValidatedNel<NumberFormatException, Int> =
+ *   if (s.matches(Regex("-?[0-9]+"))) s.toInt().validNel()
+ *   else NumberFormatException("$s is not a valid integer.").invalidNel()
+ *
+ * fun main() {
+ *   //sampleStart
+ *   val list = listOf("1", "2", "3").k().traverse(Either.applicative(), ::parseIntEither)
+ *   val failFastList = listOf("1", "abc", "3", "4s").k().traverse(Either.applicative(), ::parseIntEither)
+ *   //sampleEnd
+ *   println("list= $list")
+ *   println("failFastList= $failFastList")
+ * }
+ * ```
+ *
+ * ```kotlin:ank:playground
+ * import arrow.core.Either
+ * import arrow.core.Nel
+ * import arrow.core.ValidatedNel
+ * import arrow.core.extensions.nonemptylist.semigroup.semigroup
+ * import arrow.core.extensions.validated.applicative.applicative
+ * import arrow.core.fix
+ * import arrow.core.invalidNel
+ * import arrow.core.k
+ * import arrow.core.validNel
+ *
+ * fun parseIntEither(s: String): Either<NumberFormatException, Int> =
+ *   if (s.matches(Regex("-?[0-9]+"))) Either.Right(s.toInt())
+ *   else Either.Left(NumberFormatException("$s is not a valid integer."))
+ *
+ * fun parseIntValidated(s: String): ValidatedNel<NumberFormatException, Int> =
+ *   if (s.matches(Regex("-?[0-9]+"))) s.toInt().validNel()
+ *   else NumberFormatException("$s is not a valid integer.").invalidNel()
+ *
+ * fun main() {
+ *   //sampleStart
+ *   val validatedList = listOf("1", "22w", "3", "33s").k()
+ *     .traverse(ValidatedNel.applicative(Nel.semigroup<NumberFormatException>()), ::parseIntValidated).fix()
+ *   //sampleEnd
+ *   println(validatedList)
+ * }
+ * ```
+ *
+ * Notice that in the `Either` case, should any string fail to parse the entire [traverse] is considered a failure. Moreover, once it hits its first bad parse, it will not attempt to parse any others down the line (similar behavior would be found with using `Option`). Contrast this with `Validated` where even if one bad parse is hit, it will continue trying to parse the others, accumulating any and all errors as it goes. The behavior of [traverse] is closely tied with the [Applicative] behavior of the data type, where computations are runed in isolation.
+ *
+ * Going back to our `IO` example from the beginning with concurrency in mind, we can get an [Applicative] instance for `IO`, by using `parApplicative`. Then when we traverse a `List<A>` with its [Traverse] instance `ListK.traverse()` and a function`(A) -> IO<B>`, we can imagine the traversal as a scatter-gather. Each `A` creates a concurrent computation that will produce a `B` (the scatter), and as the `IO` operations completes they will be gathered back into a `List`.
+ *
+ * Ultimately, we use `parTraverse` that runs each operation in parallel:
```

Comment:
This sounds weird ^-^
```suggestion
 * Ultimately, we have the `parTraverse` as a shorthand for `IO` which calls `traverse` using the `parApplicative`:
```